### PR TITLE
[✨ feat] 그룹 생성 API 구현  

### PR DESCRIPTION
**Pull request by jsoonworld**
# 🚀 What’s this PR about?
- **작업 내용 요약:** 그룹 생성 API를 구현했습니다.
- **핵심 변경사항:** 초대 코드를 생성하고, 그룹을 생성하는 기능을 구현했습니다.

# 🛠️ What’s been done?
- **주요 변경사항:**
  - 요구사항에 맞게 초대 코드를 생성하는 기능을 구현했습니다.
  - 초대 코드를 포함하여 그룹을 생성하는 기능을 구현했습니다.
  - 각 기능에 대해서 테스트 코드를 작성했습니다.

# 🧪 Testing Details
- **테스트 코드 및 결과:**
  - 각 기능이 정상적으로 동작하는지 확인하는 테스트 코드를 작성하였습니다.
  - 담당자의 S3, Auth 관련 기능이 개발 완료되면 기능에 반영하여 그룹 생성 테스트를 다시 진행할 예정입니다.

# 👀 Checkpoints for Reviewers
- **리뷰 시 확인해야 할 사항:**
  - 초대 코드 생성 로직이 요구사항에 부합하는지.
  - 그룹 생성 기능이 명확하고 재사용성이 높은지.
  - 테스트 코드가 각 기능들을 올바르게 커버하고 있는지.

# 🎯 Related Issues
- #31

### CODE REVIEW

File: src/test/java/org/noostak/server/group/application/groupService/CreateGroupServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            Long userId = savedMemberId;
+            String groupName = "New Group";
+            String groupImageUrl = "https://bucket-noostak.s3.ap-northeast-2.amazonaws.com/images/group1.png";
+            GroupCreateRequest request = GroupCreateRequest.of(groupName, groupImageUrl);
+
+            // When
+            GroupCreateResponse response = groupService.createGroup(userId, request);
+
+            // Then
+            assertThat(response.groupName()).isEqualTo("New Group");
```
Comment (bbabbi) [inline src/test/java/org/noostak/server/group/application/groupService/CreateGroupServiceTest.java:65]: 지금은 그룹 이름에 대해서만 검증하고 있는데, 검증이 필요한 다른 요소가 있을지 생각해봐도 좋을 것 같네요! InviteCode에 대한 검증을 추가한다던지?

File: src/test/java/org/noostak/server/group/application/groupService/CreateGroupServiceTest.java
Comment (bbabbi) [inline src/test/java/org/noostak/server/group/application/groupService/CreateGroupServiceTest.java:1]: 혹시 그룹 이름 중복된 경우는 GroupName에서 처리해주고 있나요? 없다면 그것도 고려해주면 좋을 것 같기도 하네요!

File: src/test/java/org/noostak/server/group/application/groupService/GetAllGroupsServiceTest.java
Comment (bbabbi) [inline src/test/java/org/noostak/server/group/application/groupService/GetAllGroupsServiceTest.java:1]: 테스트코드 보다가 궁금해진 내용인데 저희 그룹 이름 불러올 때, 그룹에 join한 순서대로 불러오는 건가요? 아니면 다른 기준이 있나요?

**Review by bbabbi — APPROVED** [review]
와! 정말 고생하셨습니다
저번처럼 잘 구현해주신 부분 하나하나 코멘트 남길까 하다가 (코멘트 1n개 이래버리면) 귀찮으실 것 같아서 궁금한 부분에만 코멘트 남겼습니다.
전반적으로 코드가 깔끔해서 알아보기 쉽고, 제가 빠졌나? 싶은 부분들 없이 다 구현되어있고, 생각 못하고 넘어갈 뻔한 부분도 다 고려해서 코드 구현하신 거에서 고민 많이 하신 게 잘 느껴졌습니다!

고생하셨습니다~!~!

File: src/test/java/org/noostak/server/group/application/groupService/CreateGroupServiceTest.java (hunk 1 above)
Comment (jsoonworld) [reply src/test/java/org/noostak/server/group/application/groupService/CreateGroupServiceTest.java:65]: 피드백 감사합니다!

초대 코드 검증은 초대 코드 `VO`에서 이미 수행하도록 설계했기 때문에, `Service`에서는 해당 로직이 정상적으로 동작하는지 확인하는 데 초점을 두었습니다.

추가적으로, 테스트를 더욱 탄탄하게 하기 위해 다양한 값이나 `edge case`를 시도해보는 것도 좋은 접근이 될 것 같네요! 😊

File: src/test/java/org/noostak/server/group/application/groupService/CreateGroupServiceTest.java
Comment (jsoonworld) [reply src/test/java/org/noostak/server/group/application/groupService/CreateGroupServiceTest.java:1]: 피드백 감사합니다!

명세서에 따르면 그룹 이름은 중복이 가능하다고 되어 있어서 현재는 따로 중복 검증을 처리하지 않았습니다. 다만, 추후 요구사항이 변경될 가능성도 있으니 이를 염두에 두고 검토해보는 것도 좋은 방향일 것 같네요! 😊

File: src/test/java/org/noostak/server/group/application/groupService/GetAllGroupsServiceTest.java
Comment (jsoonworld) [reply src/test/java/org/noostak/server/group/application/groupService/GetAllGroupsServiceTest.java:1]: 피드백 감사합니다!

그룹 이름이 아니라 멤버 이름을 말씀하신 거라면, 현재 멤버 이름은 특정한 순서가 정해져 있지 않은 것으로 보입니다. 다만, 특정 정렬 기준이 필요하다면 요구사항에 따라 추가로 고려해볼 수 있을 것 같아요! 😊
